### PR TITLE
Create a client and transport per request in the TLS case to force TLS handshakes

### PR DESCRIPTION
**Pull request by pierDipi**
For testing TLS certs rotation we need to force TLS handshakes multiple times

### CODE REVIEW

**Comment by pierDipi** [comment]
@creydr @matzew the test is fixed now, PTAL

File: test/e2e/eventshub/receiver_tls_test.go [hunk 1]
```diff
@@ -83,22 +84,23 @@ func receiverTLS() *feature.Feature {
...
 	)
 	f.Assert("Sent event", assert.OnStore(sourceName).
 		MatchSentEvent(cetest.HasId(event.ID())).
-		AtLeast(1),
+		AtLeast(10),
 	)
 	f.Assert("Sender received expected peer certificate", assert.OnStore(sourceName).
 		MatchPeerCertificatesReceived(assert.MatchPeerCertificatesFromSecret(secretName, "tls.crt")).
-		AtLeast(1),
+		AtLeast(5),
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/e2e/eventshub/receiver_tls_test.go:103]: To not depend on implementation details, anything more than 1 for the test works!

[the remark this answers is not in the document]
Comment (creydr) [reply test/e2e/eventshub/receiver_tls_test.go:103]: ok. Thanks for explaining.
/lgtm

**Pull request merged**